org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/Utils.java

```diff
@@ -787,6 +792,38 @@ public static boolean isResourceAvailable(String path) {
 		return res;
 	}
 
+	/**
+	 *
+	 * @param directoryId ArtifactTemplateDirectoryId of the ArtifactTemplate that should contain a reference to a git repository.
+	 * @return The URL and the branch/tag that contains the files for the ArtifactTemplate. null if no git information is given.
+	 */
+	public static GitInfo getGitInformation(ArtifactTemplateDirectoryId directoryId) {
+		if (!(directoryId.getParent() instanceof ArtifactTemplateId)) {
+			return null;
+		}
+		RepositoryFileReference ref = BackendUtils.getRefOfDefinitions((ArtifactTemplateId)directoryId.getParent());
+		try (InputStream is = Repository.INSTANCE.newInputStream(ref)) {
+			Unmarshaller u = JAXBSupport.createUnmarshaller();
+			Definitions defs = ((Definitions) u.unmarshal(is));
+			Map<QName, String> atts = defs.getOtherAttributes();
+			String src = atts.get(new QName(Namespaces.TOSCA_WINERY_EXTENSIONS_NAMESPACE, "gitsrc"));
+			String branch = atts.get(new QName(Namespaces.TOSCA_WINERY_EXTENSIONS_NAMESPACE, "gitbranch"));
+			// ^ is the XOR operator
+			if (src == null ^ branch == null) {
+				Utils.LOGGER.error("Git information not complete, URL or branch missing");
+				return null;
+			} else if (src == null && branch == null) {
+				return null;
```

Comment:
Like I mentioned in the JavaDoc comment, if null is returned, insufficient information is given to get a Git repository. In the second case (both repo and branch are empty) it is assumed that these are missing because the user does not want to use a remote Git repo.
The first tests if only one of the two fields is empty, indicating a configuration error.